File: scm-java/src/main/scheme/primitives/PrimitiveBytevectorLength.java
package scheme.primitives;

import scheme.*;

public class PrimitiveBytevectorLength extends Primitive {
    @Override public String name() { return "bytevector-length"; }
    @Override public String info() {
        return "Syntax: (bytevector-length bv)\n" +
               "Library: (scheme base)\n" +
               "Description: Returns the number of bytes in the given bytevector.\n" +
               "Example:\n" +
               "  (bytevector-length #u8(1 2 3)) => 3";
    }

    @Override
    public Object apply(SourcePos pos, Object[] arguments) {
        checkArgs(pos, arguments, 1, 1);
        return (long) Value.asBytevector(arguments[0]).length;
    }
}

File: scm-csharp/scheme/primitives/PrimitiveBytevectorLength.cs
namespace scheme;

public class PrimitiveBytevectorLength : Primitive
{
    public override string Name() => "bytevector-length";
    public override string Info() =>
        "Syntax: (bytevector-length bv)\n" +
        "Library: (scheme base)\n" +
        "Description: Returns the number of bytes in the given bytevector.\n" +
        "Example:\n" +
        "  (bytevector-length #u8(1 2 3)) => 3";

    public override object Apply(SourcePos? pos, object[] arguments)
    {
        CheckArgs(pos, arguments, 1, 1);
        return (long)Value.AsBytevector(arguments[0]).Length;
    }
}
